server/service/src/apis/login_v4.rs

```diff
@@ -144,16 +146,26 @@ impl LoginApiV4 {
             .await
             .map_err(|e| LoginV4Error::ConnectionError(e))?;
 
-        if reqwest::StatusCode::UNAUTHORIZED == response.status() {
+        let status = response.status();
+        let body = response.json::<serde_json::Value>().await.unwrap();
+
+        if status == reqwest::StatusCode::UNAUTHORIZED {
             return Err(LoginV4Error::Unauthorised);
         }
 
-        let response = response
-            .error_for_status()
-            .map_err(|e| LoginV4Error::ConnectionError(e))?
-            .json()
-            .await
-            .map_err(|e| LoginV4Error::ConnectionError(e))?;
+        // handle account blocked error (i.e. too many failed login attempts)
+        if status == reqwest::StatusCode::FORBIDDEN {
+            if !body["timeoutRemaining"].is_null()
+                && !body["status"].is_null()
+                && body["status"] == "user_login_timeout"
+            {
+                let timeout_remaining = body["timeoutRemaining"].as_u64().unwrap();
```

Comment:
Please try to avoid `unwrap()`, see comment below...

Reply:
Was trying to avoid nested IFs plus I'd also checked `!body["timeoutRemaining"].is_null()` not null before unwrapping its value. Updated with d4921fa24